bed_file_input_modal.js

```diff
@@ -0,0 +1,219 @@
+        <span class="delete-icon">&#x2715;</span>
+    </div>
+    <div class="button-container">
+        <button class="load-button">Load</button>
+        <button class="cancel-button">Cancel</button>
+    </div>
+</dialog>
+`;
+
+class BedFileInputModal extends HTMLElement {
+    constructor() {
+        super();
+        this.attachShadow({ mode: 'open' });
+        this.shadowRoot.appendChild(modalTemplate.content.cloneNode(true));
+
+        this.modal = this.shadowRoot.querySelector('.modal');
+        this.loadButton = this.shadowRoot.querySelector('.load-button');
+        this.cancelButton = this.shadowRoot.querySelector('.cancel-button');
+        this.deleteIcons = this.shadowRoot.querySelectorAll('.delete-icon');
+
+        this.fileInput = this.shadowRoot.getElementById('fileInput');
+        this.fileInput.setAttribute('accept', '.bed');
+        this.fileInputText = this.shadowRoot.getElementById('fileInputText');
+        this.urlInput = this.shadowRoot.getElementById('urlInput');
+
+        this.addEventListeners();
+    }
+
+    addEventListeners() {
+        this.loadButton.addEventListener('click', () => this.handleLoad());
+        this.cancelButton.addEventListener('click', () => this.close());
+        this.deleteIcons.forEach((icon, index) => {
+            icon.addEventListener('click', () => this.clearInput(index));
+        });
+        this.fileInput.addEventListener('change', (e) => this.handleFileInputChange(e));
+        this.urlInput.addEventListener('input', () => this.toggleInput('url'));
+        this.fileInputText.addEventListener('click', () => this.fileInput.click());
+    }
+
+    handleFileInputChange(event) {
+        const fileName = event.target.files[0]?.name || 'Select custom bed file';
+        this.fileInputText.value = fileName;
+        this.toggleInput('file');
+    }
+
+    handleLoad() {
+        const file = this.fileInput.files[0];
+        const url = this.urlInput.value.trim();
+        
+        if (!file && !url) {
+            alert('Please provide either a local file or a URL!');
+            return;
+        }
+
+        if (file) {
+            // If a file is provided, check its extension
+            const fileExtension = file.name.split('.').pop().toLowerCase();
+            if (fileExtension !== 'bed') {
+                alert('Only files with the ".bed" extension are allowed.');
+                return;
+            }
+
+            const reader = new FileReader();
+            reader.onload = () => {
+                this.dispatchEvent(new CustomEvent('bed-file-selected', {
+                    detail: { bedText: reader.result },
+                    bubbles: true,
+                    composed: true
+                }));
+                this.close();
+            };
+            reader.readAsText(file);
+        } else if (url) {
+            this.dispatchEvent(new CustomEvent('bed-file-selected', {
+                detail: { bedUrl: url },
```

Comment:
We always want to emit the BED text, not the URL. You probably want to fetch the file here, then emit an event once it's received.

Reply:
We might be able to make a shared component for the local file modal and URL file modal, since these two are in the `bam.iobio` repo. But the bed file modal is in the `iobio-charts` repo.

Reply:
I think we would want to move all of them into iobio-charts.